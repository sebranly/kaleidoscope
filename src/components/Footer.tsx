import * as React from 'react';
import {
  AUTHOR_NAME,
  CHANGELOG_URL,
  CONTRIBUTOR_NAME,
  CONTRIBUTOR_URL,
  PROJECT_URL,
  WEBSITE_VERSION
} from '../constants/general';
import { getHeartEmoji, getRandomColor } from '../utils';

const Footer = () => {
  const [color, setColor] = React.useState(getRandomColor());

  const onMount = () => {
    const interval = setInterval(() => {
      setColor(getRandomColor());
    }, 3000);

    return () => clearInterval(interval);
  };

  React.useEffect(() => {
    onMount();
  }, []);

  return (
    <div className="App-footer">
      <div>
        Developed by{' '}
        <a href={PROJECT_URL} rel="noopener noreferrer" title="GitHub page for repository" target="_blank">
          {AUTHOR_NAME}
        </a>{' '}
        and{' '}
        <a
          href={CONTRIBUTOR_URL}
          rel="noopener noreferrer"
          title={`GitHub page for ${CONTRIBUTOR_NAME}`}
          target="_blank"
        >
          {CONTRIBUTOR_NAME}
        </a>{' '}
        with {getHeartEmoji(color)}
      </div>
      <div>This is not an official project: this is a fan project</div>
      <div>
        Website version{' '}
        <a href={CHANGELOG_URL} rel="noopener noreferrer" title="Website changelog" target="_blank">
          {WEBSITE_VERSION}
        </a>
      </div>
    </div>
  );
};

export { Footer };
