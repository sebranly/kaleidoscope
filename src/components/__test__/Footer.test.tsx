import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Footer } from '../Footer';

test('Footer', () => {
  const createComponent = () => <Footer />;

  render(createComponent());
  const author1Element = screen.getByText(/sebranly/);
  const author2Element = screen.getByText(/StephaneBranly/);

  expect(author1Element).toBeInTheDocument();
  expect(author2Element).toBeInTheDocument();
});
