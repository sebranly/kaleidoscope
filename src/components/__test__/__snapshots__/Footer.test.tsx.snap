// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer 1`] = `
NodeList [
  <div
    class="App-footer"
  >
    <div>
      Developed by
       
      <a
        href="https://github.com/sebranly/kaleidoscope"
        rel="noopener noreferrer"
        target="_blank"
        title="GitHub page for repository"
      >
        sebranly
      </a>
       
      with ðŸ’™
    </div>
    <div>
      This is not an official project: this is a fan project
    </div>
    <div>
      Website version
       
      <a
        href="https://github.com/sebranly/kaleidoscope/releases"
        rel="noopener noreferrer"
        target="_blank"
        title="Website changelog"
      >
        0.0.1
      </a>
    </div>
  </div>,
]
`;
